logstash-core/spec/api/lib/api/plugins_spec.rb

```diff
@@ -0,0 +1,57 @@
+# encoding: utf-8
+require_relative "../../spec_helper"
+require "sinatra"
+require "app/modules/plugins"
+require "logstash/json"
+
+describe LogStash::Api::Plugins do
+
+  include Rack::Test::Methods
+
+  def app()
+    described_class
+  end
+
+  before(:all) do
+    get "/"
+  end
+
+  let(:payload) { LogStash::Json.load(last_response.body) }
+
+  it "respond to plugins resource" do
+    expect(last_response).to be_ok
+  end
+
+  it "return valid json content type" do
+    expect(last_response.content_type).to eq("application/json")
+  end
+
+  context "#schema" do
+    it "return the expected schema" do
+      expect(payload.keys).to include("plugins", "total")
+      payload["plugins"].each do |plugin|
+        expect(plugin.keys).to include("name", "version")
+      end
+    end
+  end
+
+  context "#values" do
+
+    it "return totals of plugins" do
+      expect(payload["total"]).to eq(payload["plugins"].count)
+    end
+
+    it "return a list of available plugins" do
+      payload["plugins"].each do |plugin|
+        expect(plugin).to be_available?
+      end
+    end
+
+    it "return non empty version values" do
+      payload["plugins"].each do |plugin|
+        expect(plugin["version"]).not_to be_empty
```

Comment:
Hi, actually this is the way to implement it and not to check for null. We
aim to check for "something" to be returned so we check for empty.

If null is provided, test is going to fail anyway.

/purbon

On Mon, 14 Mar 2016 19:05 Pier-Hugues Pellerin, <notifications@github.com>
wrote:

> In logstash-core/spec/api/lib/api/plugins_spec.rb
> <https://github.com/elastic/logstash/pull/4794#discussion_r56047143>:
>
> >
> > +  context "#values" do
> > +
> > +    it "return totals of plugins" do
> > +      expect(payload["total"]).to eq(payload["plugins"].count)
> > +    end
> > +
> > +    it "return a list of available plugins" do
> > +      payload["plugins"].each do |plugin|
> > +        expect(plugin).to be_available?
> > +      end
> > +    end
> > +
> > +    it "return non empty version values" do
> > +      payload["plugins"].each do |plugin|
> > +        expect(plugin["version"]).not_to be_empty
>
> I think the correct assertion is be_nil since NIL wont implement empty?
>
> Failure/Error: expect({}[:hello]).not_to be_empty
> expected  to respond to `empty?`
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/pull/4794/files#r56047143>.
>